examples/misc/lib/effective_dart/design_bad.dart

```diff
@@ -0,0 +1,149 @@
+// ignore_for_file: type_annotate_public_apis, unused_element, unused_local_variable
+
+import 'dart:math' as math;
+import 'package:dartlang_examples_util/ellipsis.dart';
+import 'design_good.dart';
+
+void miscDeclAnalyzedButNotTested() {
+  (errors, monsters, subscription) {
+    // #docregion code-like-prose
+    // Telling errors to empty itself, or asking if it is?
+    if (errors.empty) {/* ... */}
+
+    // Toggle what? To what?
+    subscription.toggle();
+
+    // Filter the monsters with claws *out* or include *only* those?
+    monsters.filter((monster) => monster.hasClaws);
+    // #enddocregion code-like-prose
+
+    Iterable theCollectionOfErrors;
+    // #docregion code-like-prose-overdone
+    if (theCollectionOfErrors.isEmpty) {/* ... */}
+
+    monsters.producesANewSequenceWhereEach((monster) => monster.hasClaws);
+    // #enddocregion code-like-prose-overdone
+  };
+
+  () {
+    // #docregion cascades
+    var buffer = new StringBuffer0() //!
+        .write('one')
+        .write('two')
+        .write('three');
+    // #enddocregion cascades
+  };
+
+  () {
+    // #docregion type_annotate_public_apis
+    install(id, destination) => ellipsis;
+    // #enddocregion type_annotate_public_apis
+  };
+
+  () {
+    // #docregion func-expr-no-param-type
+    var names = people.map((Person person) {
+      return person.name;
+    });
+    // #enddocregion func-expr-no-param-type
+  };
+
+  () {
+    // #docregion avoid-dynamic
+    dynamic lookUpOrDefault(String name, Map map, dynamic defaultValue) {
```

Comment:
The example seems silly now that we have generic methods, and with type inference, you sometimes *have* to write `dynamic` to get `dynamic`.

In this example, there is no advantage to writing `dynamic`, you could write `Object` and the function works just as well.

I'm not sure the "don't write dynamic" recommendation actually applies to Dart 2.

Reply:
FYI, these versions of the pages are for 1.24.

> I'm not sure the "don't write dynamic" recommendation actually applies to Dart 2.

I've added a note to that effect in the page. We'll be revisiting the pages when creating the Dart 2 version of the site.